holidays/holiday_base.py

```diff
@@ -268,6 +282,22 @@ def __init__(
                         f"Country {self.country} does not have subdivision "
                         f"'{subdiv}'"
                     )
+
+            name = getattr(self, "country", None) or getattr(
+                self, "market", None
+            )
+            if language and name:
+                # Load translation.
+                try:
+                    translator = translation(
+                        name,
+                        languages=[language],
+                        localedir="locales",
+                    )
+                    self.tr = translator.gettext  # Replace `self.tr()`.
+                except FileNotFoundError:  # No translation found.
+                    pass
```

Comment:
We should probably issue a warning at a minimum here, if not an outright Exception

Reply:
Good point, now it raises an exception.

Reply:
Replaced with a warning after another thought.